app/controllers/questions_controller.rb

```diff
@@ -120,6 +120,15 @@ def destroy
       redirect_to(question_path(@question)) && return
     end
 
+    if @question.answer_count > 0
+      prevent = false  # Prevent based on being tied to well scored answers.
+      @question.answers.each{ |x| if x.score >= 0.5 then prevent = true; break; end }
+      if prevent
+        flash[:danger] = 'This question cannot be deleted because it has answers.'
+        redirect_to(question_path(@question)) && return
+      end
+    end
+
```

Comment:
Simpler way of doing it:

```suggestion
    if @question.answer_count > 0 && @question.answers.any? { |a| a.score >= 0.5 }
      flash[:danger] = 'This question cannot be deleted because it has answers.'
      redirect_to(question_path(@question)) && return
    end
```